prop/prop_test.go

```diff
@@ -159,3 +159,167 @@ func TestInt32(t *testing.T) {
 		t.Errorf("expected r to be int32(101), but was %#v", r)
 	}
 }
+
+func TestMany(t *testing.T) {
+	srv, err := dbus.SessionBus()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer srv.Close()
+
+	cli, err := dbus.SessionBus()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cli.Close()
+
+	propsSpec := map[string]map[string]*Prop{
+		"org.guelfey.DBus.Test": {
+			"one": {
+				"oneValue",
+				true,
+				EmitTrue,
+				nil,
+			},
+			"two": {
+				0,
+				true,
+				EmitInvalidates,
+				nil,
+			},
+		},
+	}
+	props := New(srv, "/org/guelfey/DBus/Test", propsSpec)
+
+	r := props.GetMust("org.guelfey.DBus.Test", "one")
+	if r != "oneValue" {
+		t.Errorf("expected r to be 'oneValue', but was %#v", r)
+	}
+	r = props.GetMust("org.guelfey.DBus.Test", "two")
+	if r != 0 {
+		t.Errorf("expected r to be 0, but was %#v", r)
+	}
+
+	ready := make(chan struct{})
+	c := make(chan *dbus.Signal, 1)
+	go func() {
+		err := cli.AddMatchSignal(dbus.WithMatchMember("PropertiesChanged"))
+		if err != nil {
+			t.Log(err)
```

Comment:
accessing `t` from a different Goroutine can cause a panic if the access happens after the test is done, see https://github.com/godbus/dbus/actions/runs/5049563677/jobs/9059166858?pr=368#step:6:354. You could pass the error to the main Goroutine via a channel, or add some other kind of synchronization to ensure the calls only happen before the main test Goroutine has returned. Same for the other tests.

Reply:
I hope now it is all in correct timing. The last change removes the signal from the dbus session directly after reading from the attached channel.

Thanks for being patient.